### DISCUSSION TITLE
When to use a new docker image?

### DISCUSSION THREAD

**Discussion by RkcCorian** [buanet/ioBroker.docker#210] [Questions & Answers / Fragen & Antworten] [answered]
Hello together,

since some months I'm using ioBroker via Docker Image v5.1.0. I update from time to time via command line Node.js and js-controller. Now I saw that there is a new Docker Image v5.2.0.

My question: As I'm updating via command line ioBroker what would be the benefit to use the new container version v5.2.0? Or in other words... when should I change the container to a newer version?

Many thanks!!

**Comment by schoenr79** [comment, marked as answer]
In general speaking when u facing issues with the container image it self, the base image contains critical security updates, or you're depeding on a feature in a recent release

**Reply by RkcCorian** [reply]
Updating Node.js and js-controller will not update security related parts fully?

**Reply by schoenr79** [reply]
This will pull the dependend patches, fixes for the Node.js runtime, and for the js-controller it self. But it will not update your docker base which may also could have secuirty or other issues.

Also using always the latest image will make you ensure to go with the latest version woks fine with your current configuration, in case you have to perform a disater recovery.

**Reply by RkcCorian** [reply]
UPDATE: This comment problem is resolved via ```sudo apt update``` in front
Many thanks! Another question and I hope not that stupid. With v5.1.0 I'm able to execute e.g. ```sudo apt-get install android-tools-adb android-tools-fastboot```. But with v5.2.0 I get the response below and I can't remember how I managed this with v5.1.0. I assume this is not an v5.2.0 problem?! Any idea what is missing? (I'm fresh to such Linux related stuff 🙈) Many thanks!

```
root@ioBroker_v5:/opt/iobroker# sudo apt-get install android-tools-adb android-tools-fastboo
t                                                                                           
sudo: unable to resolve host ioBroker_v5.2.0: Name or service not known                     
Reading package lists... Done                                                               
Building dependency tree... Done                                                            
Reading state information... Done                                                           
E: Unable to locate package android-tools-adb                                               
E: Unable to locate package android-tools-fastboot                                 
```

**Reply by RkcCorian** [reply]
@schoenr79 : Ahh, ok, thanks. I will change maybe later. But as I wrote above this was not the root-cause. The root-cause was that ```sudo apt update``` was missing. ```unable to resolve host ioBroker_v5.2.0: Name or service not known``` is coming but does not have a negative effect / is only cosmetic issues.
Below you see my created container names 🙈
![image](https://user-images.githubusercontent.com/79741077/140639395-cd86b2e3-2144-455c-8ae9-0a6a4d0f85b1.png)

**Reply by schoenr79** [reply]
Hey @RkcCorian  i think your hostname might be an issue. Because if u look at the [naming convention of allowed characters in hostnames](https://www.man7.org/linux/man-pages/man7/hostname.7.html), dots and undersorces arend allowed. 

So you should change your hostname in your docker run command or compose yaml file.

**Reply by schoenr79** [reply]
Hey.. so if the cosmetic problem isn't your issue anymore, is it the _apt_ command, because i don't get it what bothers you.

**Hint:** If you want to install additional packages within the container, you should configure the "PACKAGES" environment variable in the run command (`-e "PACKAGES=android-tools-adb,android-tools-fastboot"`). In that way you could ensure that the packages will be alywas installed whil provisioning a new iobroker container.

**Reply by RkcCorian** [reply]
There is a misunderstanding... my problem is resolved in the meantime. Many thanks!!

But good to know about ```-e "PACKAGES=android-tools-adb,android-tools-fastboot"```this was not know to me 😊

**Reply by schoenr79** [reply]
Here you will find more settings that can be controlled via environment variables. Check the [README (environment variables)](https://github.com/buanet/ioBroker.docker#environment-variables) section.
